service/src/main.rs

```diff
@@ -542,34 +539,39 @@ fn start_worker<E, T, D, InitializationHandler, WorkerModeProvider>(
 
 	initialization_handler.registered_on_parentchain();
 
-	// ------------------------------------------------------------------------
-	// initialize teeracle interval
-	#[cfg(feature = "teeracle")]
-	if WorkerModeProvider::worker_mode() == WorkerMode::Teeracle {
-		schedule_periodic_reregistration_thread(
-			send_register_xt,
-			run_config.reregister_teeracle_interval(),
-		);
+	match WorkerModeProvider::worker_mode() {
+		WorkerMode::Teeracle => {
+			// ------------------------------------------------------------------------
+			// initialize teeracle interval
+			#[cfg(feature = "teeracle")]
```

Comment:
The TEEracle code is not a lot, I am actually considering removing the feature gate in most places because it won't really affect the binary size, but I haven't thought much about it, this could be a preparation for #1418 or #1195.